crates/gosub_cairo/src/elements/text.rs

```diff
@@ -0,0 +1,233 @@
+    face_cache: HashMap<String, Face>,
+    /// Default font face to use when a font cannot be found
+    default_face: Face,
+}
+
+impl GosubFontContext {
+    /// Finds the face for the given family name, or returns the default face if no font is found.
+    fn find_face_family(&mut self, family: &str) -> &mut Face {
+        info!("Finding face for family: {}", family);
+
+        // See if we already got the face in cache
+        if self.face_cache.contains_key(family) {
+            info!("Face found in cache");
+            return self.face_cache.get_mut(family).expect("Face not found in cache");
+        }
+
+        // Parse the family name into a GenericFamily enum
+        let gf = GenericFamily::parse(family).unwrap_or(GenericFamily::SansSerif);
+
+        // Find all the fonts for this family
+        let fids: Vec<FamilyId> = self.font_ctx.collection.generic_families(gf).collect();
+        if fids.is_empty() {
+            info!("No family found for family: {}", family);
+            return &mut self.default_face;
+        }
+
+        // We only use the first font in the family
+        match self.font_ctx.collection.family(fids[0]) {
+            Some(f) => {
+                info!("Face found for family: {:?}", f.fonts());
+
+                // This first font can have multiple fonts (e.g. regular, bold, italic, etc.)
+                for font in f.fonts() {
+                    match &font.source().kind {
+                        SourceKind::Memory(blob) => {
+                            info!("Loading font face from memory");
+                            let rc = Rc::new(blob.data().to_vec());
+
+                            let face = self._library.new_memory_face(rc, 0).expect("Failed to load font face");
+                            self.face_cache.insert(family.to_string(), face);
+                        }
+                        SourceKind::Path(path) => {
+                            info!("Loading font face from path {}", path.to_str().expect("path to string"));
+
+                            let face = self
+                                ._library
+                                .new_face(path.to_str().expect("path to string"), 0)
+                                .expect("Failed to load font face");
+                            self.face_cache.insert(family.to_string(), face);
+                        }
+                    }
+                }
+            }
+            None => {
+                info!("No face found for family: {}", family);
+            }
+        }
+
+        &mut self.default_face
+    }
+}
+
+thread_local! {
+    /// We use a thread-local lazy static to ensure the font context is initialized once per thread
+    /// and is dropped when the thread exits. We need this because the FreeType library cannot be dropped
+    /// while any faces are still alive, so all is managed within this struct.
+    static LIB_FONT_FACE: Lazy<Mutex<GosubFontContext>> = Lazy::new(|| {
```

Comment:
Why is this a `Mutex`? A `RefCell` should work fine here, since it is a `thread_local`